pkgs/servers/dendrite/default.nix

```diff
@@ -14,6 +14,27 @@ buildGoModule rec {
 
   vendorSha256 = "sha256-sd2frDxtMd0YUVCCQGaNYU7KopZQC1Ld5OHnjttTrgA=";
 
+  subPackages = [
+    # The server as a monolith: https://github.com/matrix-org/dendrite/blob/main/docs/installation/5_install_monolith.md
+    "cmd/dendrite-monolith-server"
+    # The server as a polylith: https://github.com/matrix-org/dendrite/blob/main/docs/installation/6_install_polylith.md
+    "cmd/dendrite-polylith-multi"
```

Comment:
You may wish to link to the docs website instead of the raw markdown:

* https://matrix-org.github.io/dendrite/installation/install/monolith
* https://matrix-org.github.io/dendrite/installation/install/polylith